### ISSUE TITLE
Translate Keras blog article on autoencoders to Jupyter notebooks

### ISSUE THREAD

**Issue by lafemmephile** [lafemmephile/FemmEncoder#8]
References:
+ https://blog.keras.io/building-autoencoders-in-keras.html

**Comment by lafemmephile** [comment]
`Minimal Autoencoder`:
```python
import keras
from keras import layers
from keras.datasets import mnist
import matplotlib.pyplot as plt
import numpy as np

# This is the size of our encoded representations
encoding_dim = 32  # 784 / 32 floats -> compression of factor 24.5

# This is our input image
input_img = keras.Input(shape=(784,))

# "encoded" is the encoded representation of the input
encoded = layers.Dense(encoding_dim, activation='relu')(input_img)

# "decoded" is the lossy reconstruction of the input
decoded = layers.Dense(784, activation='sigmoid')(encoded)

# This model maps an input to its reconstruction
autoencoder = keras.Model(input_img, decoded)

# This model maps an input to its encoded representation
encoder = keras.Model(input_img, encoded)

# This is our encoded (32-dimensional) input
encoded_input = keras.Input(shape=(encoding_dim,))

# Retrieve the last layer of the autoencoder model
decoder_layer = autoencoder.layers[-1]

# Create the decoder model
decoder = keras.Model(encoded_input, decoder_layer(encoded_input))

# Configure model for per-pixel binary crossentropy loss, and the Adam optimizer
autoencoder.compile(optimizer='adam', loss='binary_crossentropy')

# Use MNIST without labels.
(x_train, _), (x_test, _) = mnist.load_data()

# Normalize between 0 and 1 and flatten 28x28 images into vectors length 784.
x_train = x_train.astype('float32') / 255.
x_test = x_test.astype('float32') / 255.
x_train = x_train.reshape((len(x_train), np.prod(x_train.shape[1:])))
x_test = x_test.reshape((len(x_test), np.prod(x_test.shape[1:])))
print(x_train.shape)
print(x_test.shape)

# Now let's train our autoencoder for 50 epochs:
autoencoder.fit(x_train, x_train,
                epochs=50,
                batch_size=256,
                shuffle=True,
                validation_data=(x_test, x_test))

# After 50 epochs, the autoencoder seems to reach a stable train/validation loss
# value of about 0.09. We can try to visualize the reconstructed inputs and the
# encoded representations. We will use Matplotlib.

# Encode and decode some digits for comparisons
encoded_imgs = encoder.predict(x_test)
decoded_imgs = decoder.predict(encoded_imgs)

# Plotting results
n = 10  # How many digits we will display
plt.figure(figsize=(20, 4))
for i in range(n):
    # Display original
    ax = plt.subplot(2, n, i + 1)
    plt.imshow(x_test[i].reshape(28, 28))
    plt.gray()
    ax.get_xaxis().set_visible(False)
    ax.get_yaxis().set_visible(False)

    # Display reconstruction
    ax = plt.subplot(2, n, i + 1 + n)
    plt.imshow(decoded_imgs[i].reshape(28, 28))
    plt.gray()
    ax.get_xaxis().set_visible(False)
    ax.get_yaxis().set_visible(False)

# Plot images
plt.show()
```

**Comment by lafemmephile** [comment]
Implemented with 8a846a18acda356b0f0f7c21b59f20eeecd01b89

**Comment by lafemmephile** [comment]
`Minimal Autoencoder`: 8a846a18acda356b0f0f7c21b59f20eeecd01b89

**Comment by lafemmephile** [comment]
Implemented with: b0f8bb329579739c4b1c2a721266b086109551ed
